tmt/cli.py

```diff
@@ -1623,3 +1623,62 @@ def completion_fish(context: Context, install: bool, **kwargs: Any) -> None:
     Setup shell completions for fish.
     """
     setup_completion('fish', install)
+
+
+@click.command(context_settings=dict(
+    ignore_unknown_options=True,
+    allow_extra_args=True,
+    ))
+@click.pass_context
+@click.option(
+    '-r', '--root', metavar='PATH', show_default=True, default='.',
+    help='Variable passed to tmt main. See `tmt --help`')
+@click.option(
+    '-v', '--verbose', count=True, default=0,
+    help='Variable passed to tmt main. See `tmt --help`')
+@click.option(
+    '--version', is_flag=True,
+    help="Variable passed to tmt main. See `tmt --help`")
+@click.option(
+    '--lint-type', default=None,
+    type=click.Choice(['tests', 'plans', 'stories']),
+    help='tmt types to lint')
+def pre_commit(
+        context: Context,
+        root: str,
+        verbose: int,
+        version: bool,
+        lint_type: Optional[str]) -> None:
+    """
+    Cli wrapper of tmt lint for
+    """
+    # Special handling for --version
+    if version:
+        main(['--version'])
+
+    # Check that .fmf/version file is present for the specific root requested
+    git_root = subprocess.check_output(
+        args=[
+            'git',
+            'rev-parse',
+            '--show-toplevel'],
+        text=True).strip()
```

Comment:
Oh fun `text=True` fails on CentOS8 :thinking:

Reply:
Yeah, I'd recommend using `tmt.utils.Command` and its `run()` instead. We have a fancy helper for "run a command", with logging and error handling, let's stick to it unless there's a good reason not to.

Reply:
Indeed, was planning to ask for help to make things consistent, forgot about it at some point

Reply:
@happz If you're still there, quick question, how can I bypass `logger` for `tmt.utils.Command`?

Reply:
> ... how can I bypass logger for tmt.utils.Command?

You don't :) Unless there's a good reason for *not* logging at all, a logger is a natural part of our lifes today, and there's one available for you, see the `Context.obj.logger`.

> This is used as a wrapper, so it doesn't need to have the same logging logic etc.

Maybe, but it's probably not that special to get an exception. As long as it does not get in your way, of course. But even `tmt-pre-commit` should honor `TMT_DEBUG` or `NO_COLOR`, for example. IIUIC, it does not have its own verbosity/debug options, everything's passed down to tmt, but the early debug & colorization controls could still be useful.

Reply:
Sorry for the long delay to get back to you. I have tried your suggestions, and `context.obj.logger` does not work because that is initialized by `main()` as `tmt.log.Logger.create(**kwargs)`. I have tried to make a similar logger for that, but I am using `ignore_unknown_options` + `allow_extra_args` to forward all cli options to `main` as is, so I do not have a `**kwargs` to forward to logger like that.

The main thing about this wrapper and why this hack was necessary is that click cli has to have the flags in the correct command/subcommand, i.e. `--root` has to be after `tmt` and before `lint`. So basically this one flattens out the cli flags to extract the ones that are necessary for `main` and pass everything else to `tmt lint` in order for `--root` to be accepted as an `args` to `pre-commit`. Maybe I can add all other flags necessary for main and create the logger accordingly, but can you instruct me on the options I need to setup for this?

Reply:
I see, I didn't realize this new command is not part of the main procession of tmt commands. Then I'd go with simple `Logger.create()` plus some extra bits, something like this:

```python
apply_colors_output, apply_colors_logging = tmt.log.decide_colorization(no_color, force_color)

logger = tmt.log.Logger.create()
logger.add_console_handler(apply_colors=apply_colors_logging)

contex.color = apply_colors_output
```

That should initialize the logger with sane defaults, and set up colorization to play nicely with the environment. We can teach the pre-commit command to accept more logging options later if needed.

Reply:
Thanks for the feedback. I believe this version is up to standards now, except for the other issue? Can you code-review again?

Reply:
Looks good to me, thanks.

Reply:
Why did we not just let the logger be created at `main`? Maybe we could have simply made a hidden flag `--pre-check` that is passed and moved all of the git commands there (if we even need that). Then the wrapper can just transparently forward the arguments.
```python
def pre_commit()
    args = [ "--pre-check", "lint" ] + sys.argv()
    if "--root" in args:
        # reorder the arguments to bring `--root` to the top
        indx = args.index("--root")
        args = args[indx:indx+2] + args[0:indx] + args[indx+2:]
    main(args)
```

@happz thoughts?

Reply:
Looks harmless :) I wonder what would bite us with this approach.